rs/sns/governance/src/governance.rs

```diff
@@ -2545,6 +2546,33 @@ impl Governance {
             })
     }
 
+    pub fn check_no_upgrades_in_progress(&self) -> Result<(), GovernanceError> {
+        let upgrade_proposals_in_progress = self.upgrade_proposals_in_progress();
+        if !upgrade_proposals_in_progress.is_empty() {
+            return Err(GovernanceError::new_with_message(
+                ErrorType::ResourceExhausted,
+                format!(
+                    "Another upgrade is currently in progress (proposal IDs {}). \
+                    Please, try again later.",
+                    upgrade_proposals_in_progress
+                        .into_iter()
+                        .map(|id| id.to_string())
+                        .collect::<Vec<String>>()
+                        .join(", ")
+                ),
+            ));
+        }
+
+        if self.proto.pending_version.is_some() {
+            return Err(GovernanceError::new_with_message(
+                ErrorType::ResourceExhausted,
+                "Upgrade lock currently acquired, not upgrading".to_string(),
```

Comment:
Indicate when that lock was acquired?

Or even have a fall back logic for unlocking?

Reply:
Outdated